--- /usr/local/etc/config/addons/www/xmlapi/index.cgi.orig
+++ /usr/local/etc/config/addons/www/xmlapi/index.cgi
@@ -127,6 +127,7 @@
     </td></tr>
     <tr><td><a href=./tokenregister.cgi?sid=$sid>tokenregister.cgi</a></td><td><b>Registers a new security access token</b><br/>
       <i>sid=string</i> - security access token id<br/>
+      <i>token=string</i> - token to register (optional)<br/>
       <i>desc=string</i> - description for new token id<br/>
     </td></tr>
     <tr><td><a href=./tokenrevoke.cgi?sid=$sid>tokenrevoke.cgi</a></td><td><b>Revokes an existing security access token</b><br/>
--- /usr/local/etc/config/addons/www/xmlapi/session.tcl.orig
+++ /usr/local/etc/config/addons/www/xmlapi/session.tcl
@@ -53,6 +53,19 @@
   return $newToken
 }
 
+proc register_own_token {newToken descr} {
+  # get tokens
+  array set tokens [get_tokens]
+
+  # add token to array
+  set tokens($newToken) $descr
+
+  # save tokens
+  save_tokens [array get tokens]
+
+  return $newToken
+}
+
 proc revoke_token tid {
   # get tokens
   array set tokens [get_tokens]
--- /usr/local/etc/config/addons/www/xmlapi/tokenregister.cgi.orig
+++ /usr/local/etc/config/addons/www/xmlapi/tokenregister.cgi
@@ -7,12 +7,18 @@
 
 if {[info exists sid] && [check_session $sid]} {
 
+  set token ""
   set desc ""
   set ise_id "-1"
   catch {
     set input $env(QUERY_STRING)
     set pairs [split $input &]
     foreach pair $pairs {
+      if {0 != [regexp "^token=(.*)$" $pair dummy val]} {
+        set token $val
+        continue
+      }
       if {0 != [regexp "^desc=(.*)$" $pair dummy val]} {
         set desc $val
 
@@ -36,12 +42,17 @@
         regsub -all {>}   $desc {_} desc
         regsub -all {%27} $desc {_} desc
         regsub -all {'}   $desc {_} desc
-        break
+        continue
       }
     }
   }
 
-  set newToken [register_token $desc]
+  if {$token != ""} {
+    set newToken [register_own_token $token $desc]
+  } else {
+    set newToken [register_token $desc]
+  }
+
   if { $newToken != "" } {
     puts "<token desc='$desc'>$newToken</token>"
   } else {
