--- /www/webui/webui.js.orig
+++ /www/webui/webui.js
@@ -23379,7 +23379,7 @@
 
 StartPage = Singleton.create(Page, {
   MAINMENU_ID: "MAINMENU_STARTPAGE",
-  downloadURLServer: "https://"+preURL+"update.homematic.com:8443/firmware/download?cmd=download",
+  downloadURLServer: "https://"+preURL+"update.homematic.com/firmware/download?cmd=download",
   fieldTestURLServer: "https://fieldtest-ccu3-update.homematic.com/firmware/download?cmd=download",
   fieldTestActive: "/etc/config/fieldTestActive",
   downloadURL : "",
@@ -27554,7 +27554,7 @@
     this.m_ccuProduct = getProduct();
     this.preURL = (this.m_ccuProduct < 3) ? "" : "ccu3-";
     this.m_product = "HM-RASPBERRYMATIC";
-    this.m_URLServer = "https://"+this.preURL+"update.homematic.com:8443";
+    this.m_URLServer = "https://"+this.preURL+"update.homematic.com";
     this.m_fieldTestURLServer = "https://fieldtest-ccu3-update.homematic.com";
 
     this.serial = homematic("CCU.getSerial");
