--- /www/config/easymodes/DIMMER_VIRTUAL_RECEIVER/KEY_TRANSCEIVER.tcl.orig
+++ /www/config/easymodes/DIMMER_VIRTUAL_RECEIVER/KEY_TRANSCEIVER.tcl
@@ -103,7 +103,7 @@
 set PROFILE_1(SHORT_ON_TIME_FACTOR) {31 range 0 - 31}
 set PROFILE_1(SHORT_ON_TIME_MODE) 0
 set PROFILE_1(SHORT_OUTPUT_BEHAVIOUR) {7 range 0 - 7}
-set PROFILE_1(SHORT_OPTICAL_SIGNAL_COLOR) {7 range 0 - 7}
+set PROFILE_1(SHORT_OPTICAL_SIGNAL_COLOR) {7 range 0 - 8}
 set PROFILE_1(SHORT_OPTICAL_SIGNAL_BEHAVIOUR) {1 range 0 - 12}
 set PROFILE_1(SHORT_PROFILE_REPETITIONS) {0 range 0 - 255}
 set PROFILE_1(SHORT_PROFILE_ACTION_TYPE) 1
--- /www/config/easymodes/DIMMER_VIRTUAL_RECEIVER/getColorElement.tcl.orig
+++ /www/config/easymodes/DIMMER_VIRTUAL_RECEIVER/getColorElement.tcl
@@ -15,6 +15,8 @@
   set options(5) "\${PURPLE}"
   set options(6) "\${YELLOW}"
   set options(7) "\${WHITE}"
+  set options(8) "\${lastValue}"
+  set options(9) "\${noModification}"
   append html [get_ComboBox options $param separate_${special_input_id}_$prn\_$pref PROFILE $param]
   append html "</td></tr>"
   return $html
--- /www/config/easymodes/js/HmIPWeeklyProgram.js.orig
+++ /www/config/easymodes/js/HmIPWeeklyProgram.js
@@ -1843,9 +1843,11 @@
       if ((this.chnType == this.DIMMER) || (this.chnType == this.UNIVERSAL_LIGHT_RECEIVER) || (this.chnType == this.SERVO)) {
         if (((this.chnType == this.DIMMER) || (this.chnType == this.UNIVERSAL_LIGHT_RECEIVER) ) && (! this.WINDOW_DRIVE_RECEIVER) ) {
           result += (val == 0) ? "<option value='0' selected='selected'>" + translateKey('optionOFF') + "</option>" : "<option value='0'>" + translateKey('optionOFF') + "</option>";
-          for (loop = 5; loop <= 100; loop += 5) {
-            optionVal = (loop / 100).toFixed(3);
-            result += (val == optionVal) ? "<option value='" + optionVal + "' selected='selected'>" + loop + " %</options>" : "<option value='" + optionVal + "'>" + loop + " %</options>";
+          for (loop = 1; loop <= 100; loop += 1) {
+            if((loop < 20) || ((loop % 5) == 0)){
+              optionVal = (loop / 100).toFixed(3);
+              result += (val == optionVal) ? "<option value='" + optionVal + "' selected='selected'>" + loop + " %</options>" : "<option value='" + optionVal + "'>" + loop + " %</options>";
+            }
           }
           result += (val == 1.005) ? "<option value='1.005' selected='selected'>" + translateKey('optionOldLevel') + "</option>" : "<option value='1.005'>" + translateKey('optionOldLevel') + "</option>";
           result += (val == 1.01) ? "<option value='1.010' selected='selected'>" + translateKey('optionNoChange') + "</option>" : "<option value='1.010'>" + translateKey('optionNoChange') + "</option>";
@@ -2052,6 +2054,7 @@
 
     result += "<select id='color_" + this.chn + "_" + this.prn + "' dataid='color_" + number + "' onchange='setColorOutputBehaviour(this.value," + this.prn + ");'>";
     result += (val == 14) ? "<option value='14' selected='selected'>" + translateKey("colorOldValue") + "</option>" : "<option value='14'>" + translateKey("colorOldValue") + "</option>";
+		result += (val == 15) ? "<option value='15' selected='selected'>" + translateKey("colorNoChange") + "</option>" : "<option value='15'>" + translateKey("colorNoChange") + "</option>";
     jQuery.each(this.arColorA, function (index, color) {
       result += (val == index) ? "<option  value='" + index + "'  selected='selected'>" + color + "</option>" : "<option  value='" + index + "'>" + color + "</option>";
     });
--- /www/rega/pages/msg/setDimmerVirtualReceiverCombinedParam.htm.orig
+++ /www/rega/pages/msg/setDimmerVirtualReceiverCombinedParam.htm
@@ -5,6 +5,7 @@
       <td>${lblColorNr}: </td>
       <td>
         <select id="combinedParam_Color">
+          <option value="254">${colorOldValue}</option>
           <option value="0">${optionColorBLACK}</option>
           <option value="1">${optionColorBLUE}</option>
           <option value="2">${optionColorGREEN}</option>
@@ -22,15 +23,33 @@
       <td>
         <select id="combinedParam_Level" >
           <option value="0">${lblOff}</option>
+          <option value="1">1%</option>
+          <option value="2">2%</option>
+          <option value="3">3%</option>
+          <option value="4">4%</option>
+          <option value="5">5%</option>
+          <option value="6">6%</option>
+          <option value="7">7%</option>
+          <option value="8">8%</option>
+          <option value="9">9%</option>
           <option value="10">10%</option>
+          <option value="15">15%</option>
           <option value="20">20%</option>
+          <option value="25">25%</option>
           <option value="30">30%</option>
+          <option value="35">35%</option>
           <option value="40">40%</option>
+          <option value="45">45%</option>
           <option value="50">50%</option>
+          <option value="55">55%</option>
           <option value="60">60%</option>
+          <option value="65">65%</option>
           <option value="70">70%</option>
+          <option value="75">75%</option>
           <option value="80">80%</option>
+          <option value="85">85%</option>
           <option value="90">90%</option>
+          <option value="95">95%</option>
           <option value="100">100%</option>
           <option value="100.5">${optionLastValue}</option>
           <option value="101">${lblIgnore}</option>
--- /www/rega/pages/msg/setRGBWCombinedParam.htm.orig
+++ /www/rega/pages/msg/setRGBWCombinedParam.htm
@@ -30,6 +30,7 @@
       <td>
         <select id="combinedParam_Level">
           <option value="0">${lblOff}</option>
+          <option value="5">5%</option>
           <option value="10">10%</option>
           <option value="20">20%</option>
           <option value="30">30%</option>
--- /www/rega/pages/msg/setRGBWTunableWhiteCombinedParam.htm.orig
+++ /www/rega/pages/msg/setRGBWTunableWhiteCombinedParam.htm
@@ -7,6 +7,7 @@
       <td>
         <select id="combinedParam_Level">
           <option value="0">${lblOff}</option>
+          <option value="5">5%</option>
           <option value="10">10%</option>
           <option value="20">20%</option>
           <option value="30">30%</option>
